platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java

```diff
@@ -169,22 +164,10 @@ void onDestroy() {
    * Called before the OnMapReadyCallback is invoked.
    */
   void onPreMapReady() {
-    invalidateCameraPosition();
```

Comment:
Actually, it might be beneficial to keep `transform.invalidateCameraPosition()` call here to always have a non-null camera position once the control is handled to the user. I believe we refactored `Transform` in the past to always, even internally, use `Transform#getCameraPostion` instead of the `cameraPosition` field and avoid NPEs (unless core returns a `null` camera position), but just to future proof, we can invalidate that position here as well.

Reply:
The use-cases are handled correctly by transform.

```

  @UiThread
  public final CameraPosition getCameraPosition() {
    if (cameraPosition == null) {
      cameraPosition = invalidateCameraPosition();
    }
    return cameraPosition;
  }
```

```
but just to future proof, we can invalidate that position here as well.
```

hardening and thus future proofing is handled in Transform.java, if there is no concrete indication to keep this use-case around. I would prefer not doing any obsolete work.

Reply:
alternatively, I'm ok with keeping that onPreMapReady hook but let's remove the hardened check in `Transform#getCameraPosition()` instead

Reply:
I'm referring to a scenario when we would use `Transform#cameraPosition` field instead of the `Transform#getCameraPosition` method to perform operations on the camera object inside the `Transform` class. All of those usages have been refactored to use `Transform#getCameraPosition`, but I'm worried they might pop up again.

However, on the second thought, I'm not really sure whether we should perform obsolete calls to harden that, tests should be the way to go. Let's remove the call now, as it's not needed, and try to expnad our test suite to cover for that.

Reply:
I think you concerns are valid. I did a find usages on the `Transfrom#cameraPosition`: 3 usages were found accessing this field directly and all where handling the null case. The more I think about it, the more I would prefer rewriting this part of the SDK..